diffkemp/simpll/DifferentialFunctionComparator.h

```diff
@@ -42,9 +45,15 @@ class DifferentialFunctionComparator : public FunctionComparator {
     /// Specific comparison of attribute lists.
     /// Attributes that do not affect the semantics of functions are removed.
     int cmpAttrs(const AttributeList L, const AttributeList R) const override;
+    /// Handle comparing of memory allocation function in cases where the size
+    /// of the composite type is different
+    int cmpOperations(const Instruction *L, const Instruction *R,
+                    bool &needToCmpOperands) const override;
 
   private:
     const DebugInfo *DI;
+
+    DataLayout *LayoutL, *LayoutR;
```

Comment:
I think it would be better to use a reference (DataLayout &) and to get the data layout from the module directly (see the note above).